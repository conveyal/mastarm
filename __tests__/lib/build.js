/* globals afterEach, beforeEach, describe, it, expect, jasmine */
const build = require('../../lib/build')
const util = require('../test-utils/util.js')

const originalTimeout = jasmine.DEFAULT_TIMEOUT_INTERVAL

describe('build', () => {
  const mockDir = util.mockDir

  beforeEach(() => {
    jasmine.DEFAULT_TIMEOUT_INTERVAL = 20000
  })
  afterEach(() => {
    jasmine.DEFAULT_TIMEOUT_INTERVAL = originalTimeout
  })

  describe('development', () => {
    it('should transform js', (done) => {
      const results = build({
        config: {},
        files: [[`${mockDir}/index.js`]]
      })

      results[0]
        .then((buf) => {
          expect(buf.toString()).toMatchSnapshot()
          done()
        })
        .catch(done)
    })

    it('should transform css', (done) => {
      const results = build({
        config: {},
        files: [[`${mockDir}/index.css`]]
      })

      results[0]
        .then((results) => {
          expect(results.css).toMatchSnapshot()
          done()
        })
        .catch(done)
    })
  })

  describe('production', () => {
    it('should transform and minify js', (done) => {
      const results = build({
        config: {},
        env: 'production',
        files: [
          [`${mockDir}/index.js`],
          [`${mockDir}/index.css`]
        ],
        minify: true
      })

      Promise.all(results).then((output) => {
        expect(output[0].toString()).toMatchSnapshot()
        expect(output[1].css).toMatchSnapshot()
        done()
      }).catch(done)
    })
  })
})
