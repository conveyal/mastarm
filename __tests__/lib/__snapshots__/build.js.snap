exports[`build development should transform css 1`] = `
".criticalClass {
  font-weight: bold !important;
}

/*# sourceMappingURL=data:application/json;base64,eyJ2ZXJzaW9uIjozLCJzb3VyY2VzIjpbIjxpbnB1dCBjc3MgMT4iXSwibmFtZXMiOltdLCJtYXBwaW5ncyI6IkFBQUE7RUFDRSw2QkFBNkI7Q0FDOUIiLCJmaWxlIjoidG8uY3NzIiwic291cmNlc0NvbnRlbnQiOlsiLmNyaXRpY2FsQ2xhc3Mge1xuICBmb250LXdlaWdodDogYm9sZCAhaW1wb3J0YW50O1xufVxuIl19 */"
`;
